homeassistant/components/shelly/__init__.py

```diff
@@ -83,59 +69,111 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
 
     coap_context = await get_coap_context(hass)
 
+    # First check if device is not sleeping
```

Comment:
Why even try connecting? The odds of it being online are so small and it halts setup of Home Assistant for `AIOSHELLY_DEVICE_TIMEOUT_SEC` seconds.

Reply:
```python
if entry.data["sleep_period"]:
    device = …
else:
    try:
        …
    except
```

Reply:
When first adding a device we reach here from `config_flow` after the user was asked to wake up the device

Reply:
This part is refactored, based on `sleep_period`:
= 0, regular device setup
= None or device not in registry, wait for device and finish setup.
else sleeping device setup